gRPC/Client Examples/EvokedPotentialsApp/BICManager.cs

```diff
@@ -168,6 +175,43 @@ public bool BICConnect()
             Console.WriteLine("Connecting to implantable device.");
             var connectDeviceReply = deviceClient.ConnectDevice(new ConnectDeviceRequest() { DeviceAddress = DeviceName, LogFileName = "./deviceLog.txt" });
 
+            // Print out impedances of electrodes
+            for (uint channelNum = 0; channelNum < numSensingChannelsDef; channelNum++)
+            {
+                bicGetImpedanceReply chanImpedValue = deviceClient.bicGetImpedance(new bicGetImpedanceRequest() { DeviceAddress = DeviceName, Channel = channelNum });
+                Console.WriteLine("CH " + (channelNum + 1).ToString() + ": " + chanImpedValue.Success);
+                if (chanImpedValue.Success == "success")
+                {
+                    // add impedance value and units to buffer
+                    impedBuffer.Add(chanImpedValue.ChannelImpedance.ToString() + " " + chanImpedValue.Units);
+                }
+                else
+                {
+                    // note error in retrieiving impedance 
+                    impedBuffer.Add(chanImpedValue.Success);
+                }
+            }
+
+            // Check that impedances can be logged
+            if (File.Exists(impedFilePath))
+            {
+                File.Delete(impedFilePath);
+            }
+            impedFileStream = new FileStream(impedFilePath, FileMode.Create, FileAccess.Write, FileShare.None, 4096, FileOptions.Asynchronous);
+            impedFileWriter = new StreamWriter(impedFileStream);
+            string impedEntry = "";
+            for (int chNum = 0; chNum < impedBuffer.Count; chNum++)
+            {
+                impedEntry = "CH" + (chNum + 1).ToString();
+                impedEntry += ", " + impedBuffer[chNum];
+                impedFileWriter.WriteLine(impedEntry);
+            }
+
+            // Close impedance logging items
+            impedFileWriter.Flush();
+            impedFileWriter.Dispose();
+            impedFileStream.Dispose();
```

Comment:
seems like based on this logging scheme we should have a timestamp and we wouldn't need to just delete a preexisting file - just have each connection add a new row and then timestamp each?